File: OpenIM-SDK/src/main/java/io/openim/android/sdk/listener/OnMsgSendCallback.java
package io.openim.android.sdk.listener;


import io.openim.android.sdk.models.Message;

/**
 * Message send listener
 */
public interface OnMsgSendCallback extends OnBase<Message> {
    /**
     * Send failed
     */
    default void onError(int code, String error){}

    /**
     * Upload progress
     */
    default void onProgress(long progress){}

    /**
     * Send succeeded
     */
    default void onSuccess(Message s){}
}

File: OpenIM-SDK/src/main/java/io/openim/android/sdk/listener/OnMsgSendCallback.kt
package io.openim.android.sdk.listener

import io.openim.android.sdk.models.Message

interface OnMsgSendCallback : OnBase<Message> {
    override fun onError(code: Int, error: String) {}
    fun onProgress(progress: Long) {}
    override fun onSuccess(data: Message) {}
}
